### PR TITLE
Added reading partitioned mesh to readMeshFromFile

### PR DESCRIPTION
**Pull request by wenqing**
As the title.

The first commit is PR #577.

### CODE REVIEW

File: MeshLib/Mesh.cpp [hunk 1]
```diff
@@ -269,5 +269,15 @@ void Mesh::setNodesConnectedByElements()
...
+std::size_t Mesh:: getMaximumNConnectedNodesToNode() const
+{
+	std::size_t max_ncn = 0;
+	for(auto node : _nodes)
+	{
+		if(node->getConnectedNodes().size() > max_ncn)
+			max_ncn = node->getConnectedNodes().size();
+	}
+	return max_ncn + 1;
+}
```
Comment (rinkk) [inline MeshLib/Mesh.cpp]: do you actually *need* this method here? I cannot find a call to this function anywhere in the code. but if you are going to need it only once or twice it makes more sense to implement it where you need it instead of enlarging the general mesh-interface. ( also, I think you could implement this more efficiently using ```std::max_element()``` )

File: MeshLib/MeshSubset.h [hunk 2]
```diff
@@ -91,6 +91,11 @@ class MeshSubset
         return _msh.getElements().cend();
     }
 
+    const Mesh &getMesh() const
+    {
+        return _msh;
+    }
```
Comment (rinkk) [inline MeshLib/MeshSubset.h]: is this method necessary? from what i gather you could access the mesh from the general mesh vector using the mesh id provided in this class? i am not sure, though.

File: MeshLib/Mesh.cpp (hunk 1 above)
Comment (wenqing) [inline MeshLib/Mesh.cpp]: Yes,  I need it for the pre-memory allocation for PETSc matrix. Its usage will be shown in  one coming PR.

File: MeshLib/MeshSubset.h (hunk 2 above)
Comment (wenqing) [inline MeshLib/MeshSubset.h]: Yes, its usage will be shown in another PR for parallel FEM with PETSc.

File: MeshLib/Mesh.cpp (hunk 1 above)
Comment (wenqing) [inline MeshLib/Mesh.cpp]: Removed and will be  added in another PR.

Comment (wenqing) [inline MeshLib/Mesh.cpp]: And it will be moved to the derived class.

File: FileIO/readMeshFromFile.cpp [hunk 3]
```diff
@@ -42,6 +55,7 @@ MeshLib::Mesh* readMeshFromFile(const std::string &file_name)
 
 	if (BaseLib::hasFileExtension("vtu", file_name))
 		return VtuInterface::readVTUFile(file_name);
+#endif
```
Comment (wenqing) [inline FileIO/readMeshFromFile.cpp]: Moved.

File: FileIO/readMeshFromFile.cpp [hunk 4]
```diff
@@ -29,11 +33,20 @@
 #include "Legacy/MeshIO.h"
 #include "FileIO/VtkIO/VtuInterface.h"
 #include "readMeshFromFile.h"
+// FileIO : for reading partitioned mesh.
+#ifdef USE_PETSC
+#include "MPI_IO/NodePartitionedMeshReader.h"
```
Comment (wenqing) [inline FileIO/readMeshFromFile.cpp]: Added.

**Comment by wenqing** [comment]
@endJunction  Removed one commit.

**Comment by norihiro-w** [comment]
:+1:
